deepchecks/checks/methodology/model_inference_time.py

```diff
@@ -0,0 +1,94 @@
+"""The model inference time check module."""
+import typing as t
+import timeit
+
+import pandas as pd
+
+from deepchecks import SingleDatasetBaseCheck, CheckResult, Dataset, ConditionResult
+from deepchecks.utils import model_type_validation
+from deepchecks.string_utils import format_number
+
+
+__all__ = ['ModelInferenceTimeCheck']
+
+
+MI = t.TypeVar('MI', bound='ModelInferenceTimeCheck')
+
+
+class ModelInferenceTimeCheck(SingleDatasetBaseCheck):
+    """Measure model average inference time (in seconds) per sample.
+
+    Args:
+        number_of_samples (int): number of samples to use for inference
```

Comment:
maximum number of samples (dataset might be smaller)

Reply:
updated